src/code/z_demo.c

```diff
@@ -151,14 +151,14 @@ void func_800645A0(PlayState* play, CutsceneContext* csCtx) {
     Input* input = &play->state.input[0];
 
     if (CHECK_BTN_ALL(input->press.button, BTN_DLEFT) && (csCtx->state == CS_STATE_IDLE) &&
-        (gSaveContext.sceneSetupIndex >= 4)) {
+        (gSaveContext.sceneLayer > SCENE_LAYER_ADULT_NIGHT)) {
```

Comment:
Would it be worth having a `SCENE_LAYER_CUTSCENES_FIRST` (?) name for a 4 here, instead of relying on `SCENE_LAYER_ADULT_NIGHT` being the last non-cutscene layer?

Reply:
I had the same thought yeah. Just wasnt sure what an elegant thing to call it would be, or if it was even worth.

Reply:
I'm starting to think it would be worth it, it would be better to understand what those checks are, and since all such checks have been `< 4`-like it seems to have been the natural thing to do so far

It could also be used in the main layer computation
`gSaveContext.sceneLayer = 4 + (gSaveContext.cutsceneIndex & 0xF);`

Reply:
fwiw not all of them were like that, but a majority of them were. it was whatever the person decompiling felt like putting :D 
but yeah, im just not sure what to call it

Reply:
`#define SCENE_LAYER_CUTSCENE_BASE SCENE_LAYER_ADULT_NIGHT + 1`
idk, maybe others will have more ideas

Reply:
Well, I was thinking you'd whack a `!` in front of it. but yes, fundamentally it comes down to what makes more sense in context.

Reply:
for some reason thought it wouldnt match like that, but it seems to so far. will try out the change and see how it looks

Reply:
I like the way it turned out, pushed change

Reply:
think i would probably prefer a name like `SCENE_LAYER_IS_CUTSCENE`, but i can also understand wanting to keep it like the other IS_ macros
in general i find `IS_` unfortunate cause it doesnt indicate a clear system that the macro pertains to